src/client/CustomerClientApi.ts

```diff
@@ -0,0 +1,34 @@
+import { Customer, CustomersApi, SearchCustomersRequest } from 'square-connect';
+import { IFindOrCreateCustomerRequest } from '../interface';
+
+export class CustomerClientApi extends CustomersApi {
+    async findOrCreateCustomer(findOrCreateCustomerRequest: IFindOrCreateCustomerRequest): Promise<Customer> {
+        const { phone_number }: IFindOrCreateCustomerRequest = findOrCreateCustomerRequest;
```

Comment:
email also needs to be required, if you do not have it - just create a customer via `createCustomer` method